src/thunderbots/software/ai/hl/stp/play/halt_play.cpp

```diff
@@ -0,0 +1,49 @@
+#include "ai/hl/stp/play/halt_play.h"
+
+#include "ai/hl/stp/play/play_factory.h"
+#include "ai/hl/stp/tactic/stop_tactic.h"
+
+const std::string HaltPlay::name = "Halt Play";
+
+std::string HaltPlay::getName() const
+{
+    return HaltPlay::name;
+}
+
+bool HaltPlay::isApplicable(const World &world) const
+{
+    return world.gameState().isHalted();
+}
+
+bool HaltPlay::invariantHolds(const World &world) const
+{
+    return world.gameState().isHalted();
```

Comment:
Can you also add gamestate checks to the other existing plays to make sure that they are _not_ applicable if the game state is not correct?

Reply:
Will do in a separate PR